spec/system/admin/order_spec.rb

```diff
@@ -172,6 +172,27 @@ def new_order_with_distribution(distributor, order_cycle)
     expect(order.line_items.reload.map(&:product)).to include product
   end
 
+  context "When adding a product on an order with transaction fee" do
+    let(:order_with_fees) { create(:completed_order_with_fees, user:, distributor:, order_cycle: ) }
+
+    it 'recalculates transaction fee' do
+      login_as_admin
+      visit spree.edit_admin_order_path(order_with_fees)
+
+      transaction_fee = order_with_fees.all_adjustments.payment_fee.eligible.first.amount
+      expect(page.find("#order_adjustments").text).to have_content(transaction_fee)
+
+      select2_select product.name, from: 'add_variant_id', search: true
+      find('button.add_variant').click
+      sleep(1)
+
```

Comment:
You should be able to remove the `sleep` by checking the variant is added on the page.